plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/ProjectStatePersister.java

```diff
@@ -301,14 +308,11 @@ private void writeFingerprints(Collection<? extends HashedFileContent> files, Da
 		}
 	}
 
-	private void writeValidationIssues(Map<URI, ? extends Collection<Issue>> validationIssues, DataOutput output)
-			throws IOException {
-
+	private void writeValidationIssues(Multimap<URI, Issue> validationIssues, DataOutput output) throws IOException {
 		int numberSources = validationIssues.size();
 		output.writeInt(numberSources);
-		for (Map.Entry<URI, ? extends Collection<Issue>> srcIssues : validationIssues.entrySet()) {
-			URI source = srcIssues.getKey();
-			Collection<Issue> issues = srcIssues.getValue();
+		for (URI source : validationIssues.keys()) {
```

Comment:
Due to the change from `Map<URI,Collection<Issue>>` to `MultiMap<URI,Issue>` this for-loop's block won't be executed for resources without issues.

However, because the original code converted the map to an `ListMultiMap` and with `MultiMap#asMap()` back to a Map in method `#writeProjectState()`, it has the same effect if this is the only call-path that leads here. But double check if not done already.

Reply:
Thanks, I agree. Also, I couldn't find indications for writing out empty lists to the project state.